projects/batfish/src/main/java/org/batfish/z3/ReachabilityQuerySynthesizer.java

```diff
@@ -163,6 +173,23 @@ public NodProgram getNodProgram(NodProgram baseProgram) throws Z3Exception {
     queryConditions.addConjunct(finalActions);
     queryConditions.addConjunct(SaneExpr.INSTANCE);
 
+    // check transit constraints (unordered)
+    if (_transitNodes != null && _transitNodes.size() > 0) {
```

Comment:
rule of thumb I'm trying to put in: no variable that's not `@Nullable` can be `null`. So many ways to fix here:

- constructor parameter is not `@Nullable`
- constructor is `@Nullable`, but we default to empty in constructor so that `_transitNodes` is not `@Nullable`
- `_transitNodes` is `@Nullable`.

I lean towards first or second alternative. Because then we can drop the `if` altogether -- the list will always be present (maybe empty). AFAICT, the `.size() > 0` check is not needed, because it'll just be a `for` loop that terminates immediately.